Java/core/src/main/java/com/amazon/randomcutforest/executor/PointStoreCoordinator.java

```diff
@@ -16,58 +16,43 @@
 package com.amazon.randomcutforest.executor;
 
 import static com.amazon.randomcutforest.CommonUtils.checkNotNull;
-import static com.amazon.randomcutforest.CommonUtils.toFloatArray;
 
 import java.util.List;
 import java.util.Optional;
 
-import com.amazon.randomcutforest.store.PointStore;
+import com.amazon.randomcutforest.state.store.PointStoreDoubleState;
 import com.amazon.randomcutforest.store.PointStoreDouble;
-import com.amazon.randomcutforest.store.PointStoreDoubleData;
 
 public class PointStoreCoordinator implements IUpdateCoordinator<Integer> {
 
-    private final PointStoreDouble doubleStore;
-    private final PointStore store;
+    private final PointStoreDouble store;
 
     public PointStoreCoordinator(PointStoreDouble store) {
         checkNotNull(store, "store must not be null");
-        this.doubleStore = store;
-        this.store = null;
-    }
-
-    public PointStoreCoordinator(PointStore store) {
-        checkNotNull(store, "store must not be null");
-        this.doubleStore = null;
         this.store = store;
     }
 
     @Override
     public Integer initUpdate(double[] point, long seqNum) {
```

Comment:
I'm open to this idea, but I prefer the current names. The important thing about these methods isn't just that they are called before and after `update` is called on each `IUpdatable`. `initUpdate` produces the value that's consumed in the `update` calls, and `completeUpdate` takes the result of the update calls. So `initUpdate` and `completeUpdate` are really part of the forest update.

That said, do you find the names confusing? What if we changed initUpdate to beginUpdate? Do you still prefer preUpdate and postUpdate?

Reply:
I am a skeptical about "postUpdate" -- the "post processing" is a part of the update because otherwise the state will not remain meaningful. Its "postParallel" but not postUpdate ...